pl_examples/intel_compressor/imagenet.py

```diff
@@ -0,0 +1,88 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example is largely adapted from https://github.com/pytorch/examples/blob/master/imagenet/main.py.
+
+Before you can run this example, you will need to download the ImageNet dataset manually from the
+`official website <http://image-net.org/download>`_ and place it into a folder `path/to/imagenet`.
+
+Train on ImageNet with default parameters:
+
+.. code-block: bash
+
+    python imagenet.py --data-path /path/to/imagenet
+
+or show all options you can change:
+
+.. code-block: bash
+
+    python imagenet.py --help
+"""
+from argparse import ArgumentParser, Namespace
+
+import pytorch_lightning as pl
+from pl_examples import cli_lightning_logo
+from pl_examples.domain_templates.imagenet import ImageNetLightningModel
+
+
+def main(args: Namespace) -> None:
+    if args.seed is not None:
+        pl.seed_everything(args.seed)
+
+    print(args)
+
+    assert args.default_root_dir is not None, "Please set the path to save the quantized model"
+
+    if args.accelerator == "ddp":
+        # When using a single GPU per process and per
+        # DistributedDataParallel, we need to divide the batch size
+        # ourselves based on the total number of GPUs we have
+        args.batch_size = int(args.batch_size / max(1, args.gpus))
+        args.workers = int(args.workers / max(1, args.gpus))
+
+    model = ImageNetLightningModel(**vars(args))
+
+    if args.evaluate:
+        # load and validate quantized model
+        import neural_compressor
+
+        model.model = neural_compressor.utils.pytorch.load(args.default_root_dir, model.model)
+        trainer = pl.Trainer.from_argparse_args(args)
+        trainer.test(model)
+        return
+
+    quantizer_cb = pl.callbacks.INCQuantization(
+        "config/quantization.yaml", metric_name="val_acc1", save_path=args.default_root_dir
+    )
+    trainer = pl.Trainer.from_argparse_args(args, callbacks=[quantizer_cb])
+
+    trainer.compressor(model)
+
+
+def run_cli():
+    parent_parser = ArgumentParser(add_help=False)
+    parent_parser = pl.Trainer.add_argparse_args(parent_parser)
+    parent_parser.add_argument("--data-path", metavar="DIR", type=str, help="path to dataset")
+    parent_parser.add_argument("--seed", type=int, default=42, help="seed for initializing training.")
+    parent_parser.add_argument(
+        "-e", "--evaluate", dest="evaluate", action="store_true", help="evaluate model on validation set"
+    )
+    parser = ImageNetLightningModel.add_model_specific_args(parent_parser)
+    parser.set_defaults(profiler="simple", deterministic=True, max_epochs=90)
+    args = parser.parse_args()
+    main(args)
```

Comment:
Let's use the LightningCLI. Sync with @carmocca @mauvilsa if you are experiencing issues.

Reply:
Hi,@carmocca, Could you tell me where is the LightningCLI example? 
I just refer pl_examples/domain_templates/imagenet.py here.

Reply:
@PenghuiCheng probably you should first have a quick look at the documentation https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_cli.html. Regarding examples, there are multiple which use LightningCLI, see the search in the repo https://github.com/PyTorchLightning/pytorch-lightning/search?l=Python&q=LightningCLI.

Reply:
Done